packages/js/src/utils/errors.ts

```diff
@@ -34,13 +34,20 @@ const getMessageByErrorCode = (errorCode: LogtoErrorCode): string => {
 
 export class LogtoError extends Error {
   code: LogtoErrorCode;
-  error?: string;
-  errorDescription?: string;
+  data: unknown;
 
-  constructor(code: LogtoErrorCode, error?: string, errorDescription?: string, message?: string) {
-    super(message ?? getMessageByErrorCode(code));
+  constructor(code: LogtoErrorCode, data?: unknown) {
+    super(getMessageByErrorCode(code));
+    this.code = code;
+    this.data = data;
+  }
+}
+
+export class LogtoRequestError extends Error {
+  code: string;
+
+  constructor(code: string, message?: string) {
+    super(message ?? '');
```

Comment:
should fallback to a meaningful string, e.g. `Logto request failed.`

Reply:
Fixed, PTAL.

Reply:
Updated. `message` is required and does not have default value any more. PTAL